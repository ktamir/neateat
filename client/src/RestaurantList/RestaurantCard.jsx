// @flow

import { Card } from 'antd';
import React from 'react';
import './RestaurantCard.scss';
import type { Restaurant } from '../flowTypes';

type Props = {
  restaurant: Restaurant
}

export const RestaurantCard = (props: Props) => {
  const { name, accepts10Bis, rating } = props.restaurant;
  return (
    <Card className="restaurant-card">
      <div className="card-details">
        <h3>{name}</h3>
        {accepts10Bis && <span className="ten-bis">10bis</span>}
        <div>Rating: {Array.from({ length: rating }, (_, i) =>
          <span aria-label="star" role="img" key={i}>‚≠ê</span>)}</div>
      </div>
    </Card>
  );
};
