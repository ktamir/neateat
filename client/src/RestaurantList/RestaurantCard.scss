.restaurant-card {
  width: 300px;
  height: 100px;
  h3 {
    display: inline;
  }
  margin: 15px;
  .ten-bis {
    color: gray;
    font-size: 9px;
    display: inline;
    margin-left: 5px;
  }
  .card-details {
    margin-left: 20px;
  }
}
