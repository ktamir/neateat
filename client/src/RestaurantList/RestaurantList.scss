 .restaurant-fetch-error {
    margin-top: 20px;
  }
