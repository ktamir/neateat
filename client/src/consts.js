// @flow

export const API_URL = 'http://localhost:3000';

// To be retrieved from an API, etc...
export const cuisines = [
  {
    key: 'italian',
    displayName: 'Italian',
  },
  {
    key: 'french',
    displayName: 'French',
  },
  {
    key: 'burgers',
    displayName: 'Burgers',
  },
];

export const FETCH_RESTAURANTS_REQUEST = 'FETCH_RESTAURANTS_REQUEST';

export const FETCH_RESTAURANTS_SUCCESS = 'FETCH_RESTAURANTS_SUCCESS';

export const FETCH_RESTAURANTS_ERROR = 'FETCH_RESTAURANTS_ERROR';

// Used in test snapshots
export const mockRestaurants = [
  {
    id: 1,
    name: 'Chez Tamir',
    cuisine: 'French',
    rating: 2,
    address: '144 Derech Menachem Begin, Tel Aviv, Israel',
    maxDeliveryTime: 60,
    accepts10Bis: true,
  },
  {
    id: 2,
    name: 'Tamir Trattoria',
    cuisine: 'Italian',
    rating: 1,
    address: '7 Derech Namir, Tel Aviv, Israel',
    maxDeliveryTime: 100,
    accepts10Bis: true,
  },
];
