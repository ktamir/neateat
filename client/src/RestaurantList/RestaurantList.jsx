// @flow

import React, { Component } from 'react';
import { RestaurantCard } from './RestaurantCard';
import type { Restaurant } from '../flowTypes';
import { fetchRestaurants } from '../store/restaurantActions';
import { connect } from 'react-redux';
import { Spin } from 'antd';

type Props = { restaurants: Array<Restaurant>, isLoading: boolean, fetchRestaurants: Function };

export class RestaurantList extends Component<Props> {
  componentDidMount(): void {
    this.props.fetchRestaurants();
  }

  render() {
    const { isLoading, restaurants } = this.props;
    return (
      <div className="restaurant-list">
        {
          isLoading ? <Spin tip="Loading..." /> :
            restaurants.map(restaurant =>
              <RestaurantCard key={restaurant.id} restaurant={restaurant} />)
        }
      </div>
    );
  }
}

const mapStateToProps = state => state;

const mapDispatchToProps = dispatch => ({ fetchRestaurants: () => dispatch(fetchRestaurants) });

export const RestaurantListContainer = connect(mapStateToProps, mapDispatchToProps)(RestaurantList);
