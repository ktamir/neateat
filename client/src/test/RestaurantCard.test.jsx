import { configure, shallow } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16/build';
import { mockRestaurants } from '../consts';
import { RestaurantCard } from '../RestaurantList/RestaurantCard';

configure({ adapter: new Adapter() });

describe('RestaurantCard', () => {
  it('should render correctly', () => {
    let mockRestaurant = mockRestaurants[0];
    let restaurantCard = shallow(<RestaurantCard restaurant={mockRestaurant} />);
    expect(restaurantCard).toMatchSnapshot();
  });
});
