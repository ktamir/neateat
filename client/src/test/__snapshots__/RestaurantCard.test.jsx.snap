// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantCard
    restaurant={
      Object {
        "accepts10Bis": true,
        "address": "144 Derech Menachem Begin, Tel Aviv, Israel",
        "cuisine": "French",
        "id": 1,
        "maxDeliveryTime": 60,
        "name": "Chez Tamir",
        "rating": 2,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="card-details"
      >
        <h3>
          Chez Tamir
        </h3>
        <span
          className="ten-bis"
        >
          10bis
        </span>
        <div>
          Rating: 
          <span
            aria-label="star"
            role="img"
          >
            ⭐
          </span>
          <span
            aria-label="star"
            role="img"
          >
            ⭐
          </span>
        </div>
      </div>,
      "className": "restaurant-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Chez Tamir
          </h3>,
          <span
            className="ten-bis"
          >
            10bis
          </span>,
          <div>
            Rating: 
            <span
              aria-label="star"
              role="img"
            >
              ⭐
            </span>
            <span
              aria-label="star"
              role="img"
            >
              ⭐
            </span>
          </div>,
        ],
        "className": "card-details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Chez Tamir",
          },
          "ref": null,
          "rendered": "Chez Tamir",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "10bis",
            "className": "ten-bis",
          },
          "ref": null,
          "rendered": "10bis",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Rating: ",
              Array [
                <span
                  aria-label="star"
                  role="img"
                >
                  ⭐
                </span>,
                <span
                  aria-label="star"
                  role="img"
                >
                  ⭐
                </span>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "Rating: ",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "aria-label": "star",
                "children": "⭐",
                "role": "img",
              },
              "ref": null,
              "rendered": "⭐",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "aria-label": "star",
                "children": "⭐",
                "role": "img",
              },
              "ref": null,
              "rendered": "⭐",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="card-details"
        >
          <h3>
            Chez Tamir
          </h3>
          <span
            className="ten-bis"
          >
            10bis
          </span>
          <div>
            Rating: 
            <span
              aria-label="star"
              role="img"
            >
              ⭐
            </span>
            <span
              aria-label="star"
              role="img"
            >
              ⭐
            </span>
          </div>
        </div>,
        "className": "restaurant-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Chez Tamir
            </h3>,
            <span
              className="ten-bis"
            >
              10bis
            </span>,
            <div>
              Rating: 
              <span
                aria-label="star"
                role="img"
              >
                ⭐
              </span>
              <span
                aria-label="star"
                role="img"
              >
                ⭐
              </span>
            </div>,
          ],
          "className": "card-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Chez Tamir",
            },
            "ref": null,
            "rendered": "Chez Tamir",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "10bis",
              "className": "ten-bis",
            },
            "ref": null,
            "rendered": "10bis",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Rating: ",
                Array [
                  <span
                    aria-label="star"
                    role="img"
                  >
                    ⭐
                  </span>,
                  <span
                    aria-label="star"
                    role="img"
                  >
                    ⭐
                  </span>,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              "Rating: ",
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "aria-label": "star",
                  "children": "⭐",
                  "role": "img",
                },
                "ref": null,
                "rendered": "⭐",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "aria-label": "star",
                  "children": "⭐",
                  "role": "img",
                },
                "ref": null,
                "rendered": "⭐",
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
